tests/test_basic.py

```diff
@@ -344,3 +344,106 @@ def test_rbd_pvc(rook_cluster):
     rook_cluster.kubernetes.kubectl('delete pod csirbd-demo-pod')
     rook_cluster.kubernetes.kubectl('delete pvc rbd-pvc')
     rook_cluster.kubernetes.kubectl('delete sc rook-ceph-block')
+
+
+def set_storage_pattern(rook_cluster, pattern):
+    # create a working-copy of cluster.yaml
+    cluster_working = os.path.join(rook_cluster.ceph_dir, 'cluster_test.yaml')
+    cluster_original = os.path.join(rook_cluster.ceph_dir, 'cluster.yaml')
+
+    cluster_orig_f = open(cluster_original, 'r')
+    cluster_object = yaml.full_load(cluster_orig_f)
+
+    cluster_object['spec']['storage']['deviceFilter'] = pattern
+
+    cluster_work_f = open(cluster_working, 'a')
+    cluster_work_f.write(yaml.dump(cluster_object, Dumper=yaml.Dumper))
+
+    cluster_orig_f.close()
+    cluster_work_f.close()
+
+    return cluster_working
+
+
+def test_storage_pattern(rook_cluster):
+    # cleanup basic installation
+    cluster_original = os.path.join(rook_cluster.ceph_dir, 'cluster.yaml')
+    rook_cluster.kubernetes.kubectl('delete -f ' + cluster_original)
+    i = 0
+    while rook_cluster.get_number_of_osds() != 0:
+        if i == 10:
+            pytest.fail("There are still OSDs up and running")
+        logger.info("Still osds running")
+        time.sleep(10)
+        i += 1
+
+    # add a disk of 10 G the node
+    nodes = rook_cluster.kubernetes.hardware.workers
+    disk_name = nodes[0].disk_create(10)
+    nodes[0].disk_attach(name=disk_name)
+
+    # get the device name of the new volume
+    volume_name = nodes[0].get_volume_name(disk_name)
+    # set storage pattern to cluster.yaml
+    # set device filter to e.g. ^vdc
+    pattern = "^" + re.split(r'/', volume_name)[-1]
+    cluster_new = set_storage_pattern(rook_cluster, pattern)
+
+    # install the rook-cluster
+    rook_cluster.install_ceph(cluster_new, False)
+
+    # re-start rook operator to get new osd-prepare jobs
+    operator = rook_cluster.get_operator_pod()
+    rook_cluster.kubernetes.kubectl('-n rook-ceph delete pod ' + operator)
+    # wait for new operator pod
+    while operator == rook_cluster.get_operator_pod():
+        time.sleep(15)
+
+    logger.info("Wait for exactly one OSD to run ")
+    pattern = re.compile(r'.*rook-ceph-osd-*')
+    common.wait_for_result(
+        rook_cluster.kubernetes.kubectl,
+        "--namespace rook-ceph get deployments",
+        matcher=common.regex_exact_matcher(pattern, 1),
+        attempts=90, interval=10)
+
+    # get back to basic installation
+    rook_cluster.install_ceph()
+
+
+def test_storage_full_path_pattern(rook_cluster):
+    # cleanup basic installation
+    cluster_original = os.path.join(rook_cluster.ceph_dir, 'cluster.yaml')
+    rook_cluster.kubernetes.kubectl('delete -f ' + cluster_original)
```

Comment:
Since we know these tests are back-to-back we could skip doing the install_ceph in the previous test. However, because they would be tightly coupled I think we should just test both scenarios as on test method (as in merge `test_storage_full_path_pattern` and `test_storage_pattern` into a generic `test_storage_patterns`)

Reply:
Good point.
"Merged" as one single test.